pkg/log/zap/zap.go

```diff
@@ -245,7 +250,8 @@ func (o *Options) BindFlags(fs *flag.FlagSet) {
 		o.StacktraceLevel = fromFlag
 	}
 	fs.Var(&stackVal, "zap-stacktrace-level",
-		"Zap Level at and above which stacktraces are captured (one of 'warn' or 'error')")
+		"Zap Level at and above which stacktraces are captured (one of 'info', 'error', or 'disabled')."+
+			"--zap-stacktrace-level=disabled will set level to be 'fatal'.")
```

Comment:
wdyt about the warn level? Should we not keep this one?

Reply:
@joelanford @bharathi-tenneti Agreed, Let's defer on the changes to `--zap-stacktrace-level=disabled`  and follow up in a separate PR.

It's more important that we fix the `--zap-log-levels` bug without going back and forth on the semantics of how to disable stacktraces for now. We can circle back to this.

@bharathi-tenneti We can just keep the levels `info` and `error` for disabling stacktraces for now.

@camilamacedo86 No point in keeping `warn`. 
With logr, users can log at `info`/`logr.V(0).Info()` or `error`/`logr.Error()`. There's no level for `warn`.

`--zap-stacktrace-level=warn` would capture stacktraces for `warn, error, dpanic, ...` and not for `debug, info`.
And `--zap-stacktrace-level=error` would capture stacktraces for `error, dpanic, ...` and not for `debug, info, warn`.

So `error` should suffice.

Reply:
Hi @hasbro17, 

It is fine and I agree that it can be a pr to fix and then a pr to add the flag `"--zap-stacktrace-level=disabled`. But then we need to : 
- Not  add "--zap-stacktrace-level=disabled` in this PR
- Add a //todo comment describing what should be done in a follow-up and track an issue for that in this repo.  

c/c @bharathi-tenneti